### ISSUE TITLE
CredoLanguageServer.handle_request/2 not handling TextDocumentDocumentSymbol

### ISSUE THREAD

**Issue by wkirschbaum** [elixir-tools/credo-language-server#8]
When hooking up to Emac's lsp program called eglot, the server crashes when trying to handle the following request:

```elixir
%GenLSP.Requests.TextDocumentDocumentSymbol{
  params: %GenLSP.Structures.DocumentSymbolParams{
    partial_result_token: nil,
    work_done_token: nil,
    text_document: %GenLSP.Structures.TextDocumentIdentifier{
      uri: "file:///home/whk/src/elixir/za_id_number/lib/za_id_number/validator.ex"
    }
  },
  id: 3,
  jsonrpc: "2.0",
  method: "textDocument/documentSymbol"
}
```

The following change fixes it for me locally:

```diff
@@ -104,6 +104,10 @@ defmodule CredoLanguageServer do
     {:noreply, assign(lsp, exit_code: 0)}
   end
 
+  def handle_request(_, lsp) do
+    {:noreply, lsp}
+  end
+
   @impl true
   def handle_notification(%Initialized{}, lsp) do
     GenLSP.log(lsp, "[Credo] LSP Initialized!")
```

**Comment by mhanberg** [comment]
So the server is not advertising that it supports the document symbol capability, so the client should not be sending the request.

**Comment by wkirschbaum** [comment]
Sure, but when is it a good idea to trust a client to not crash the server?

**Comment by wkirschbaum** [comment]
This is eglot in emacs, will have a look there too.

**Comment by mhanberg** [comment]
https://github.com/mhanberg/.dotfiles/blob/20316645d2eac5d8c2435de86e2c3fe495acb091/config/nvim/lua/motch/autocmds.lua#L53

Example 👆

**Comment by mhanberg** [comment]
> Sure, but when is it a good idea to trust a client to not crash the server?


Fair point, but I'm just saying the protocol specifies these capabilities just for this reason and most other LSP clients seem to handle it fine.

**Comment by mhanberg** [comment]
Let me take another look at the protocol to see what it says about how to handle a situation like this.

**Comment by mhanberg** [comment]
I think I know the change I need to make, I'll push that up shortly

**Comment by wkirschbaum** [comment]
I will submit a patch to emacs for this if i can spot the bug there, but thanks for having a look.

**Comment by mhanberg** [comment]
I am not sure about your config but could possibly make sure you don't have a plug-in that is the culprit.

**Comment by wkirschbaum** [comment]
This is core emacs with no strange config options. I have not seen this issue with other language servers I use.

**Comment by mhanberg** [comment]
@wkirschbaum would you mind trying out the branch on #15 and let me know if it works?

**Comment by wkirschbaum** [comment]
@mhanberg thanks. I am not getting the same error, but have timeouts after 30 seconds. This was observed before on larger projects, but happening on smaller ones on this branch as well. I will investigate a bit more and log an issue a bit later.

**Comment by mhanberg** [comment]
@wkirschbaum Would you be able to share with me a minimal emacs eglot config to get me started? Would probably help figure this out faster

**Comment by wkirschbaum** [comment]
For testing I use the following: 

```elisp
(require 'eglot)

(setq eglot-server-programs `((elixir-ts-mode . ("~/src/elixir/credo-language-server/bin/credo-language-server"))))
;; or
(setq eglot-server-programs `((elixir-ts-mode . ("localhost" 9000))))

(add-hook 'elixir-ts-mode-hook 'eglot-ensure)
(add-hook 'heex-ts-mode-hook 'eglot-ensure)
````

you can use it with elixir-mode as well from melpa and would look like this:

```elisp
(require 'eglot)

(setq eglot-server-programs `((elixir-ts-mode . ("localhost" 9000))))

(add-hook 'elixir-mode-hook 'eglot-ensure)
```

I am currently using only core emacs on the master branch, compiled with tree-sitter. 

Unless you are comfortable with Emacs I dont want to burden you with debugging, I will get to it soonish :)

**Comment by mhanberg** [comment]
@wkirschbaum i'm calling the original issue fixed by #15.

Please open a new issue once you narrow down the new problem you're having with eglot.

Thanks again!
